### DISCUSSION TITLE
Beginner - addLineTool is undefine? and tslib error

### DISCUSSION THREAD

**Discussion by Tandyku** [difurious/lightweight-charts-line-tools#8] [Q&A] [answered]
Hi, I am getting this error: TypeError: Cannot read properties of undefined (reading 'addLineTool').

const HorizontalLineTool = chart.addLineTool('HorizontalLine', [hlPoint], {			

I keep getting this error too, it doesnt specify where in my code that this error is, I assume it is in another file. 
TypeError: Failed to resolve module specifier "tslib". Relative references must start with either "/", "./", or "../"

Thanks.

**Comment by difurious** [comment]
@Tandyku you need to show more of your code.  how is the chart created?  Look at the debug.html to see creation of a basic chart and it uses similar code to what you are doing.

**Reply by Tandyku** [reply]
My bad. Here's some more code:
The chart:
chart = LightweightCharts.createChart(document.getElementById('chart'), {
                        width: 800,
                        height: 488,
                        timeScale: {
                            timeVisible: true,
                            barSpacing: 20 // Adjust bar spacing to fit all data points
                        }
                    });

I copied the same method for creating a horizontal line:

                    
                    const displayHorizontalLine = true;
                    
                    // HorizontalLine
                    if (displayHorizontalLine) {
                    	const hlPrice = 30000.0;
                    	const hlTS = Date.parse('2024-06-03');
                    	const hlPoint = {price: hlPrice, timestamp: hlTS};	
                    
                    	const HorizontalLineTool = chart.addLineTool('HorizontalLine', [hlPoint], {			
                    		"text": {
                    			"value": "HorizontalLine Line Tool",
                    			"alignment": "left",
                    			"font": {
                    				"color": "rgba(41,98,255,1)",
                    				"size": 20,
                    				"bold": false,
                    				"italic": false,
                    				"family": "Arial"
                    			},
                    			"box": {
                    				"alignment": {
                    					"vertical": "top",
                    					"horizontal": "left"
                    				},
                    				"angle": 0,
                    				"scale": 1,
                    				"offset": {
                    					"x": 0,
                    					"y": 0
                    				},
                    				"padding": {
                    					"x": 0,
                    					"y": 0
                    				},
                    				"maxHeight": 100,
                    				"shadow": {
                    					"blur": 0,
                    					"color": "rgba(255,255,255,1)",
                    					"offset": {
                    						"x": 0,
                    						"y": 0
                    					}
                    				},
                    				"border": {
                    					"color": "rgba(126,211,33,0)",
                    					"width": 1,
                    					"radius": 0,
                    					"highlight": false,
                    					"style": 0
                    				},
                    				"background": {
                    					"color": "rgba(199,56,56,0)",
                    					"inflation": {
                    						"x": 0,
                    						"y": 0
                    					}
                    				}
                    			},
                    			"padding": 0,
                    			"wordWrapWidth": 0,
                    			"forceTextAlign": false,
                    			"forceCalculateMaxLineWidth": false
                    		},
                    		"line": {
                    			"color": "rgba(41,98,255,1)",
                    			"width": 1,
                    			"style": 0,
                    			"join": "round",
                    			"cap": "square",
                    			"end": {
                    				"left": 0,
                    				"right": 0
                    			},
                    			"extend": {
                    				"right": true,
                    				"left": true
                    			}
                    		},
                    		"visible": true,
                    		"editable": true
                    	
                    	});
                    }

But I do not see this on my chart. And I get two errors:
Uncaught TypeError: Failed to resolve module specifier "tslib". Relative references must start with either "/", "./", or "../". (This is at Index:1 for some reason)
And
index:2986 Error fetching data: TypeError: chart.addLineTool is not a function
I dont really know why it doesnt detect addLineTool as a function.

**Comment by difurious** [comment]
@Tandyku  does the debug.html work fine?  you need to make the chart, add a series, add data to the series, then addLineTool to the chart.  Follow along with the debug.html.  Just scrape the code from the .html for those parts, then see if you can get a bare min chart working.

**Reply by Tandyku** [reply]
Yes the debug.html works, my lightweight charts works well with candles and a candleseries. But still cant detect addLineTool as a function. So I think it may be something outside of the code? Like one of the other files.

So from the files that I downloaded, I only added the files from the lib/prod/src/model directory into my website's home directory. Maybe this is incorrect?

**Reply by difurious** [reply, marked as answer]
wait. all you need to do is import  'lightweight-charts.standalone.production.js'.  That is the compiled version of linetools.  Im not sure what you are doing with anything else.  I have a feeling all this is going to be over your skill level as a beginner.

**Reply by Tandyku** [reply]
Oh wow it works now, thank you man!
